stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessagesViewModelFactory.kt

```diff
@@ -54,6 +56,7 @@ public class MessagesViewModelFactory(
     private val showDateSeparators: Boolean = true,
     private val showSystemMessages: Boolean = true,
     private val deletedMessageVisibility: DeletedMessageVisibility = DeletedMessageVisibility.ALWAYS_VISIBLE,
+    private val messageFooterVisibility: MessageFooterVisibility = MessageFooterVisibility.WithTimeDifference()
 ) : ViewModelProvider.Factory {
 
     /**
```

Comment:
Seems like cohesion here is quite low - not many parameters are shared between different ViewModels. Might be a sign to split them some day, WDYT?

Reply:
Most of them are related to the `MessageList` that's true and we could split them up. But I do like the idea of having to create just one factory rather than two or three and it being responsible for all of the messages ViewModels.